### PR TITLE
Change Benchmarks to use BenchmarkDotNet

### LINKED ISSUE #8842: Benchmarks: Rework Benchmark Infra
`BenchmarkAttribute` is a `FactAttribute` but it is used on tests with parameters & alongwith `DataAttribute` like `BenchmarkVariation` which are generally used on `Theory`. As always xUnit does not like our mess and shows errors. 😝 

This errors/warnings are explicitly disabled in csproj files at present. Rework this infrastructure and remove nowarns.

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #8842

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -5,5 +5,6 @@
     <add key="AspNetCore" value="https://dotnet.myget.org/F/aspnetcore-ci-dev/api/v3/index.json" />
     <add key="AspNetCoreTools" value="https://dotnet.myget.org/F/aspnetcore-tools/api/v3/index.json" />
     <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
+    <add key="bdn-nightly" value="https://ci.appveyor.com/nuget/benchmarkdotnet" />
```
Comment (smitpatel) [inline NuGet.config]: @pranavkm @ryanbrandenburg - This is feed for BenchmarkDotNet nightly packages. We need their nightly build for a critical fix till they push a new package to NuGet.

File: build/dependencies.props [hunk 2]
```diff
@@ -20,4 +19,13 @@
     <XunitVersion>2.3.0-*</XunitVersion>
     <XunitVersionInSpecProjects>2.2.0</XunitVersionInSpecProjects>
   </PropertyGroup>
+<!-- Following package versions must match with what is available on NuGet -->
+  <PropertyGroup>
+    <BenchmarkDotNetVersion>0.10.8.*</BenchmarkDotNetVersion>
```
Comment (smitpatel) [inline build/dependencies.props]: https://github.com/aspnet/EntityFramework/issues/9285

File: build/dependencies.props [hunk 3]
```diff
@@ -20,4 +19,13 @@
...
+    <EF6Version>6.1.3</EF6Version>
+    <EFCore1Version>1.1.2</EFCore1Version>
+    <EFCore1SqlClientVersion>4.3.1</EFCore1SqlClientVersion>
+    <EFCore2Version>2.0.0-preview2-final</EFCore2Version>
```
Comment (smitpatel) [inline build/dependencies.props]: https://github.com/aspnet/EntityFramework/issues/9286

File: NuGet.config (hunk 1 above)
Comment (AndreyAkinshin) [inline NuGet.config]: BenchmarkDotNet v0.10.9 (with fix for https://github.com/dotnet/BenchmarkDotNet/issues/481) has been released, it's already available on nuget.org: https://www.nuget.org/packages/BenchmarkDotNet/0.10.9

**Pull request merged**